src/components/Callback.tsx

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useState } from "react";
+import { UserManager, User } from "oidc-client-ts";
+
+type Props = {
+  authenticated: boolean | null;
+  setAuth: (authenticated: boolean | null) => void;
+  userManager: UserManager;
+  handleLogout: any;
+};
+
+const Callback = ({
+  authenticated,
+  setAuth,
+  userManager,
+
+  handleLogout,
+}: Props) => {
+  const [userInfo, setUserInfo] = useState<User | null>(null);
+
+  useEffect(() => {
+    if (authenticated === null) {
+      userManager.signinRedirectCallback().then((user: any) => {
+        if (user) {
+          //saving the tokens to local storage
+          localStorage.setItem("access_token", user.access_token)
```

Comment:
we also must save the tokens expiration in order to know when to refresh them.

and just a small note,
I would put the whole token pair to the key like 
```
localStorage.setItem("access_token", {
  accessToken: user.access_token,
  refreshToken: user.refresh_token,
  expiresIn: user.expires_in,
})
```